server/src/main/java/org/elasticsearch/common/logging/ChunkedLoggingStream.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * An {@link OutputStream} which Gzip-compresses the written data, Base64-encodes it, and writes it in fixed-size chunks to a logger. This
+ * is useful for debugging information that may be too large for a single log message and/or which may include data which cannot be
+ * recorded faithfully in plain-text (e.g. binary data or data with significant whitespace).
+ */
+public class ChunkedLoggingStream extends OutputStream {
+
+    static final int CHUNK_SIZE = ByteSizeUnit.KB.toIntBytes(2);
```

Comment:
This functionality only really makes sense for debugging, and I don't see a great reason to want to change this.

Reply:
I've thought about this a little more and I now see value in adding some configurability here. Particularly the `NodesHotThreadsRequest` we use has some hard-coded parameters & defaults that deserve to be exposed as settings, and if we're adding those settings then why not this one too?

I won't do it in this PR tho, nor do I expect to get to it any time soon, but if you're interested then feel free to open a follow-up.

Reply:
I think 2KiB is reasonable. And since now we have some quick commands to reconstruct it, I think it is fine. Originally I had thought (before I realized of the econding) that I could have set this to something large, so that there would be one big block of human-readable text. But since now either way we need to process the encoded text, I do not mind the chunks. We can make this configurable if the need ever arises.